wdl2cwl/main.py

```diff
@@ -36,784 +21,766 @@
-        command_check = False
-    else:
-        raw_command = raw_command[
-            raw_command.find("{") + 1 : -1
-        ]  # removing the command{} part
-
-    command = textwrap.dedent(raw_command)
-
-    command = get_command(
-        command,
-        ast.task_inputs,
-        ast.task_inputs_bound,
-        input_types,
-        input_names,
-        command_check,
-    )
-
-    base_command = ["bash", "example.sh"]
-
-    cwl_inputs: List[cwl.CommandInputParameter] = []
-    cwl_inputs = get_input(cwl_inputs, ast.task_inputs, ast.task_inputs_bound)
-
-    requirements: List[cwl.ProcessRequirement] = []
-
-    if "docker" in ast.task_runtime:
-        dockerPull = ""
-
-        if '"' not in ast.task_runtime["docker"]:
-            # only searching for value in bound inputs.
-            # value could be a bound declaration which is not handled
-            for sublist in ast.task_inputs_bound:
-                if ast.task_runtime["docker"] in sublist[1]:
-                    dockerPull = sublist[2]
-
+        return js_str
+
+    def get_expr(self, wdl_expr: Any) -> str:
+        """Translate WDL Expressions."""
+        if isinstance(wdl_expr, WDL.Expr.Apply):
+            return self.get_expr_apply(wdl_expr)
+        elif isinstance(wdl_expr, WDL.Expr.Get):
+            return self.get_expr_get(wdl_expr)
+        elif isinstance(wdl_expr, WDL.Expr.IfThenElse):
+            return self.get_expr_ifthenelse(wdl_expr)
+        elif isinstance(wdl_expr, WDL.Expr.Placeholder):
+            return self.translate_wdl_placeholder(wdl_expr)
+        elif isinstance(wdl_expr, WDL.Expr.String):
+            return self.get_expr_string(wdl_expr)
+        elif isinstance(wdl_expr, WDL.Expr.Ident):
+            return self.get_expr_ident(wdl_expr)
+        elif isinstance(wdl_expr, WDL.Tree.Decl):
+            return self.get_expr(wdl_expr.expr)
+        elif isinstance(
+            wdl_expr,
+            (
+                WDL.Expr.Boolean,
+                WDL.Expr.Int,
+                WDL.Expr.Float,
+                WDL.Expr.Array,
+            ),
+        ):
+            return self.get_literal_name(wdl_expr)
         else:
-            dockerPull = ast.task_runtime["docker"]
+            raise Exception(f"The expression '{wdl_expr}' is not handled yet.")
+
+    def get_literal_name(
+        self,
+        expr: Union[
+            WDL.Expr.Boolean,
+            WDL.Expr.Int,
+            WDL.Expr.Float,
+            WDL.Expr.Array,
+        ],
+    ) -> str:
+        """Translate WDL Boolean, Int or Float Expression."""
+        if expr is None or not hasattr(expr, "parent"):
+            raise Exception(f"{type(expr)} has no attribute 'parent'")
+        # if the literal expr is used inside WDL.Expr.Apply
+        # the literal value is what's needed
+        if isinstance(expr.parent, WDL.Expr.Apply):  # type: ignore
+            return self.get_wdl_literal(expr.literal)  # type: ignore
```

Comment:
And if the parent isn't a `WDL.Expr.Apply`, what then? (This is caught by `mypy`: https://github.com/common-workflow-lab/wdl-cwl-translator/runs/4885522368?check_suite_focus=true#step:7:17 )